libsnark/gadgetlib1/gadgets/basic_gadgets.tcc

```diff
@@ -261,29 +261,16 @@ void dual_variable_gadget<FieldT>::generate_r1cs_witness_from_bits()
 template<typename FieldT>
 void disjunction_gadget<FieldT>::generate_r1cs_constraints()
 {
-    /* inv * sum = output */
-    linear_combination<FieldT> a1, b1, c1;
-    a1.add_term(inv);
-    for (size_t i = 0; i < inputs.size(); ++i) {
-        b1.add_term(inputs[i]);
-    }
-    c1.add_term(output);
+    linear_combination<FieldT> sum = pb_sum(inputs);
```

Comment:
What's the benefit to pass the inputs as a `pb_linear_combination_array`, beyond the fact that we now don't need to "manually build" the "sum" linear combination? In which case will `inputs` be an array of length > 1 of linear comb?

Reply:
> What's the benefit to pass the inputs as a pb_linear_combination_array, beyond the fact that we now don't need to "manually build" the "sum" linear combination? 

I might be missing something... Where you say "manually build" the sum near this part of the code, I assume you are talking about having removed the explict adding of terms here?  I simply replaced the very verbose longer code segment with a short version that uses the automatic conversion and the pb_sum convenience function.  This code just looked like it was written before the convenience constructors / operators / functions were written.  It should be a bit clearer now, and consist of less code to maintain.

Regarding the `pb_linear_combination_array`, that's independent of the code here.  Using linear combinations here is more versatile than variables (if we wanted to pass a linear combination in as a variable, we would have to allocate a new variable and create the constraint to make it equal the linear combination, all of which adds to the circuit size and proving time).

> In which case will inputs be an array of length > 1 of linear comb?

Regarding the specific use-case, I think an earlier version of the scalar multiplication gadget (or rather one of the gadgets it uses internally) was using it, but was then changed to no longer require it.  But hopefully the point about avoiding extra variable allocations and constraints justifies the change without an explicit use case.

Reply:
Let me know if that doesn't answer your question.  I realise now that you might have meant something else by `"maunally build" the sum`.

Reply:
> Where you say "manually build" the sum near this part of the code, I assume you are talking about having removed the explict adding of terms here?

Yes

> Regarding the specific use-case, I think an earlier version of the scalar multiplication gadget (or rather one of the gadgets it uses internally) was using it, but was then changed to no longer require it. But hopefully the point about avoiding extra variable allocations and constraints justifies the change without an explicit use case.

I see, thanks